### PR TITLE
CCMSG-1115: Add a dropdown menu to setting the data stream type config.

### PR DESCRIPTION
**Pull request by amuamushu**
## Problem
In confluent platform, there are specific values a user can choose from for setting the `data stream type`. However, currently, this configuration is set by filling in the blank which requires users to review the config description to remember what the values are and which configs have specific values.

## Solution
Add a recommender to each of those configurations such that it recommends the specific values.

##### Does this solution apply anywhere else?
- [ ] yes
- [x] no

## Test Strategy
Added the recommenders and checked in confluent platform that those configs had the correct recommenders.

##### Testing done:
- [ ] Unit tests
- [ ] Integration tests
- [ ] System tests
- [x] Manual tests

## Release Plan

Releasing as part of the data stream feature.

Not backporting or merging to master.
 
Not reverting or rolling back.

### CODE REVIEW

File: pom.xml
Comment (levzem) [inline pom.xml:166]: this is a propietary package which we cannot use in public repos

File: src/main/java/io/confluent/connect/elasticsearch/ElasticsearchSinkConnectorConfig.java
Comment (levzem) [inline src/main/java/io/confluent/connect/elasticsearch/ElasticsearchSinkConnectorConfig.java:344]: nice for fixing it for other values too!

File: src/main/java/io/confluent/connect/elasticsearch/ElasticsearchSinkConnectorConfig.java
Comment (levzem) [inline src/main/java/io/confluent/connect/elasticsearch/ElasticsearchSinkConnectorConfig.java:475]: we don't need to put this in static variables, we can just do it inline
```suggestion
            Recommenders.enumValues(...)
```

**Comment by amuamushu** [comment]
This PR is currently on hold until #532 is merged.

**Review by levzem — APPROVED** [review]

**Pull request merged**
